database/CoreDatabase.java

```diff
@@ -566,14 +572,14 @@ long committedEventCount() {
         }
     }
 
-    class StatisticsCorrector {
+    public class StatisticsCorrector {
```

Comment:
1. Make the class and some of its methods public so that other packages can access them.
2. Make some other fields and methods `protected` so that its subclasses can access them.